arm-insights/2014-04-01/swagger/insightsManagementClient_Components.json

```diff
@@ -0,0 +1,343 @@
+          }
+        }
+      },
+      "get": {
+        "description": "Returns an Application Insights instance",
+        "operationId": "Insights_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request when retrieving an Application Insights instance",
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsResource"
+            },
+            "examples": {
+              "application/json": {
+                "value": [
+                  {
+                    "id": "/subscriptions/d386467e-33de-44ff-b324-641549c30383/resourceGroups/sots-rg/providers/microsoft.insights/components/tha-demo2",
+                    "name": "tha-demo2",
+                    "type": "Microsoft.Insights/components",
+                    "location": "West Europe",
+                    "tags": {},
+                    "kind": "web",
+                    "properties": {
+                      "Ver": "v2",
+                      "ApplicationId": "tharvey-demo2",
+                      "AppId": "00000000-0000-0000-0000-000000000000",
+                      "Application_Type": "web",
+                      "Flow_Type": null,
+                      "Request_Source": null,
+                      "InstrumentationKey": "00000000-0000-0000-0000-000000000000",
+                      "Name": "tha-demo2",
+                      "CreationDate": "2016-11-22T20:42:42.299Z",
+                      "PackageId": null,
+                      "TenantId": "00000000-0000-0000-0000-000000000000",
+                      "HockeyAppId": null,
+                      "HockeyAppToken": null,
+                      "provisioningState": "Succeeded"
+                    }
+                  }
+                ],
+                "nextLink": "https://management.azure.com/########-####-####-####-############$skiptoken=######"
+              }
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/ApplicationInsightsResource"
+      },
+      "put": {
+        "description": "Creates or Updates an Application Insights instance",
+        "operationId": "Insights_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
```

Comment:
Perhaps I'm mis-understanding / you've posted this comment in the wrong place in the file - but why would the DELETE HTTP Call need a body to update?

With regards for Updating Redis instances - we're doing this via the CreateOrUpdate method - see [here for the implementation within Terraform](https://github.com/hashicorp/terraform/pull/10326/files#diff-f82489832a3be9189bfe0f08df4d2089R88) - this is hooked up and working (generated from before it was turned into a composite swagger)